modules/data-entry/src/main/frontend/src/questionnaire/IdentifierQuestion.jsx

```diff
@@ -0,0 +1,118 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useEffect, useState } from "react";
+
+import { Button, Tooltip } from "@mui/material";
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+import { v4 as uuidv4 } from 'uuid';
+
+import PropTypes from "prop-types";
+
+import Answer from "./Answer";
+import Question from "./Question";
+import AnswerComponentManager from "./AnswerComponentManager";
+
+// Component that renders an identifier question, with optional copy button.
+// Selected answers are placed in a series of <input type="hidden"> tags for
+// submission.
+//
+// Optional props:
+//  text: String containing text to show alongside the identifier
+//  displayMode: "plain", "plain+copy", or undefined denoting how the identifier is presented
+//             to the user. "+copy" adds a button the user can press to copy the value to clipboard.
+//             Defaults to "plain".
+//
+// Sample usage:
+// <IdentifierQuestion
+//    text="Visit Identifier"
+//    />
+
+export default function IdentifierQuestion(props) {
+  const { existingAnswer, pageActive, isEdit, ...rest} = props;
+  const {
+    displayMode = "plain",
+    identifierType = "uuid"
+  } = {...props.questionDefinition };
+  const [ text, setText ] = useState("Copy to Clipboard")
+
+  const [value, setValue] = useState(existingAnswer?.[1]?.value || "");
+  const answer = [["value", value]];
```

Comment:
What is the meaning of the first position of the inner array ("value")?

Reply:
The `[1]` is the answer node.
Existing value has `[0]` as the nodes key/name (a UUID) and then `[1]` as the actual node object. As a result the magic `existingAnswer?.[1]` is quite common in question.jsx files

Reply:
After a quick look at MultipleChoice, it would appear that position [0] of the answer passed to the <Answer> component might generally be the label. If that's the case, it might make more sense to pass this instead:

```suggestion
  const answer = [[value, value]];
```